tests/aws/services/lambda_/test_lambda_performance.py

```diff
@@ -81,6 +103,520 @@ def invoke():
+
+    # wait to complete the event invoke without causing extra load through polling
+    LOG.info("Waiting for event invokes to be processed ...")
+    time.sleep(200)
+
+    LOG.info("Validate invocations")
+    s3_request_ids = get_s3_keys(aws_client, s3_bucket)
+    assert set(s3_request_ids) == set(request_ids)
+
+
+@markers.aws.unknown
+def test_lambda_event_invoke(create_lambda_function, s3_bucket, aws_client, aws_client_factory):
+    """Test concurrent Lambda event invokes and validate the number of Lambda invocations using CloudWatch and S3."""
+    num_invocations = 130
+
+    function_name = f"echo-func-{short_uid()}"
+    create_lambda_function(
+        handler_file=TEST_LAMBDA_PYTHON_S3_INTEGRATION,
+        func_name=function_name,
+        runtime=Runtime.python3_12,
+        Environment={"Variables": {"S3_BUCKET_NAME": s3_bucket}},
+    )
+
+    # Limit concurrency to avoid resource bottlenecks. This is typically not required because the ThreadPoolExecutor
+    # in the pollers are limited to 32 threads. The actual concurrency depends on the number of available CPU cores.
+    # aws_client.lambda_.put_function_concurrency(
+    #     FunctionName=function_name, ReservedConcurrentExecutions=1
+    # )
+
+    lock = threading.Lock()
+    request_ids = []
+    error_counter = ThreadSafeCounter()
+    invoke_barrier = threading.Barrier(num_invocations)
+
+    def invoke(runner: int):
+        nonlocal request_ids
+        nonlocal error_counter
+        nonlocal invoke_barrier
+        try:
+            payload = {"file_size_bytes": 1}
+            lambda_client = aws_client_factory(config=CLIENT_CONFIG).lambda_
+            invoke_barrier.wait()
+            result = lambda_client.invoke(
+                FunctionName=function_name,
+                InvocationType=InvocationType.Event,
+                Payload=to_bytes(json.dumps(payload)),
+            )
+            request_id = result["ResponseMetadata"]["RequestId"]
+            with lock:
+                request_ids.append(request_id)
+        except Exception as e:
+            LOG.error(f"runner-{runner} failed: {e}")
+            error_counter.increment()
+
+    start_time = datetime.utcnow()
+    # Use threads to invoke Lambda function in parallel
+    thread_list = []
+    for i in range(1, num_invocations + 1):
+        thread = threading.Thread(target=invoke, args=[i])
+        thread.start()
+        thread_list.append(thread)
+
+    for thread in thread_list:
+        thread.join()
+    end_time = datetime.utcnow()
+    diff = end_time - start_time
+    LOG.info(f"N={num_invocations} took {diff.total_seconds()} seconds")
+    assert error_counter.counter == 0
+
+    # Sleeping here is a bit hacky, but we want to avoid polling for now because polling affects the results.
+    sleep_seconds = 200
+    LOG.info(f"Sleeping for {sleep_seconds} ...")
+    time.sleep(sleep_seconds)
+
+    # Validate CloudWatch invocation metric
+    def assert_cloudwatch_metric():
```

Comment:
@steffyP @pinzon Is this calculation of the CloudWatch metric `Invocations` correct?
I'm getting numbers that are way off (see assert below)

What's the status of v2? Does it support such a metric and query?

Reply:
simple queries like this one should work :) but actually they should also work the current implementation aka v1.

Reply:
Thanks @steffyP 

The main issue was a too-optimistic waiting time and `end_time`. [This code change](https://github.com/localstack/localstack/pull/9917/commits/42d2dc9cea1896e52d1a3c64e7a055a041ea4991#diff-c626c358a191e4cc10999a6ed5f301e2f5a7eca9d379e91731268810803e3a8cR291) made the metric more reliable.